### PR TITLE
feature: supported to use ipv6 address match route.

### PR DESCRIPTION
**Pull request by membphis**

### CODE REVIEW

**Review by c9s — APPROVED** [review]

File: tests/check_remote_addr.c [hunk 1]
```diff
@@ -77,16 +77,117 @@ START_TEST (test_remote_addrs)
...
+    entry->remote_addr.len = sizeof("fe88:fe80::1") - 1;
+    matched_route = r3_tree_match_route(n, entry);
+    ck_assert(matched_route == NULL);
+
+    entry = match_entry_create("/goo");
+    entry->remote_addr.base = "::1";
+    entry->remote_addr.len = sizeof("::1") - 1;
+    matched_route = r3_tree_match_route(n, entry);
+    ck_assert(matched_route != NULL);
+    ck_assert(matched_route->data == &uri2);
```
Comment (c9s) [inline tests/check_remote_addr.c:173]: please try to separate test cases if you're testing different test cases.

Comment (membphis) [reply tests/check_remote_addr.c:173]: Sorry, I don't understand your question. The current test case for ipv6 has been summarized into `test_remote_addrs_ipv6`.

Or do you think this part of the use case should be written in an independent test file?

**Comment by membphis** [comment]
@c9s do you have time to review this PR again?

**Comment by membphis** [comment]
can we merge this pr?

**Comment by c9s** [comment]
Thank you!

**Pull request merged**